--- a/src/gles/device.rs
+++ b/src/gles/device.rs
@@ -549,9 +549,12 @@
                 log::trace!("Get binding {name:?} from program {program:?}");
                 match register {
                     super::BindingRegister::UniformBuffers => {
-                        let index = unsafe { gl.get_uniform_block_index(program, name) }.unwrap();
-                        log::trace!("\tBinding slot {slot} to block index {index}");
-                        unsafe { gl.uniform_block_binding(program, index, slot as _) };
+                        if let Some(index) = unsafe { gl.get_uniform_block_index(program, name) } {
+                            log::trace!("\tBinding slot {slot} to block index {index}");
+                            unsafe { gl.uniform_block_binding(program, index, slot as _) };
+                        } else {
+                            log::debug!("Uniform block {name:?} was optimized out of the linked program");
+                        }
                     }
                     super::BindingRegister::StorageBuffers => {
                         let index =
